client/redux/cart/actions.js

```diff
@@ -0,0 +1,137 @@
+import {
+  TOGGLE_CART,
+  UPDATE_CART,
+  FETCH_CART_PRODUCTS_FAILURE,
+  FETCH_CART_PRODUCTS_SUCCESS,
+  FETCH_CART_PRODUCTS_REQUEST,
+  UPDATE_CART_ID,
+} from './constants'
+
+import * as api from '../../api/cart'
+
+export const toggleCart = () => ({
+  type: TOGGLE_CART,
+})
+
+export const updateCart = (cart) => ({
+  type: UPDATE_CART,
+  payload: cart,
+})
+
+export const fetchCartProductsRequest = () => ({
+  type: FETCH_CART_PRODUCTS_REQUEST,
+})
+
+export const fetchCartProductsFailure = (error) => ({
+  type: FETCH_CART_PRODUCTS_FAILURE,
+  payload: error,
+})
+
+export const fetchCartProductsSuccess = (products) => ({
+  type: FETCH_CART_PRODUCTS_SUCCESS,
+  payload: products,
+})
+
+export const getCartProducts = () => async (dispatch, getState) => {
+  const { id: cartId } = getState().cart
+  dispatch(fetchCartProductsRequest())
+
+  try {
+    const data = await api.getCart(cartId)
+
+    const { products, total } = await data.data
+
+    const quantity = products.reduce((accum, product) => (accum += product.quantity), 0)
+
+    dispatch(fetchCartProductsSuccess({ products, total, quantity }))
+  } catch (error) {
+    dispatch(fetchCartProductsFailure(error.message))
+  }
+}
+
+export const createCart = (productId) => async (dispatch) => {
+  try {
+    await api.createCart({})
```

Comment:
@AStashevskaya When I'm trying to add pizza to cart, it looks like you are using one request to create cart: `POST /api/carts` which returns `id` and sets `cookie` and then another request for fetching created cart `GET /api/carts/:cart_id`, its not required, because we have all required info in first request answer.

![extra-request](https://user-images.githubusercontent.com/12561302/116720145-285cc600-a9dc-11eb-9fe3-54abde161cc2.gif)

Also, after adding first product cart view on `UI` should be changed, but I see `0` items in cart